src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -694,7 +694,7 @@ def _do_upgrade(self):
                 to_upgrade.append(d_entry)
 
                 # if we don't have a list of others to consider, stop now
-                if not known_ok_to_stop:
+                if not known_ok_to_stop and d.daemon_type in ['osd', 'mds', 'mon']:
```

Comment:
can you change the order of the expression?

```suggestion
                if d.daemon_type in ['osd', 'mds', 'mon'] and not known_ok_to_stop:
```

cause checking for `known_ok_to_stop` only makes if we're talking about osds, mdss or mons. And *not* vice versa.